### PR TITLE
ISPN-9397 Check TX support for remote caches

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.jboss.org/browse/ISPN-9397

### CODE REVIEW

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java [hunk 1]
```diff
@@ -512,4 +512,9 @@ default ServerStatistics stats() {
     * Return the currently {@link DataFormat} being used.
     */
    DataFormat getDataFormat();
+
+   /**
+    * @return {@code true} if the cache can participate in a transaction, {@code false} otherwise.
+    */
+   boolean supportsTransactions();
```
Comment (tristantarrant) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java]: isTransactional() ?

Comment (pruivo) [reply client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java]: done

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCacheContainer.java [hunk 2]
```diff
@@ -125,4 +125,9 @@
    boolean switchToDefaultCluster();
 
    Marshaller getMarshaller();
+
+   /**
+    * @return {@code true} if the cache with name {@code cacheName} can participate in transactions.
+    */
+   boolean supportsTransactions(String cacheName);
```
[the remark this answers is not in the document]
Comment (pruivo) [reply client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCacheContainer.java]: done!

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/Util.java [hunk 3]
```diff
@@ -54,4 +79,22 @@ protected static RuntimeException rewrap(ExecutionException e) {
...
+   public static boolean checkTransactionSupport(String cacheName, OperationsFactory factory, Log log) {
+      PrepareTransactionOperation op = factory.newPrepareTransactionOperation(DUMMY_XID, true, Collections.emptyList(),
+            false, 60000);
+      try {
+         return op.execute().handle((integer, throwable) -> {
+            if (throwable != null) {
+               log.invalidTxServerConfig(cacheName, throwable);
+            }
+            return throwable == null;
+         }).get();
```
Comment (galderz) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/Util.java]: Wait should not be forever, there should be a time boundary and if exceeded return `false`?

Comment (pruivo) [reply client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/Util.java]: It doesn't wait forever. A scheduled task is registered after `sock-timeout` milliseconds and it cancels the operation.

**Pull request merged**